k8s/upgrade/src/bin/upgrade-job/helm/chart.rs

```diff
@@ -94,6 +172,43 @@ impl CoreValues {
     }
 }
 
+/// This is used to deserialize the yaml object agents.
+#[derive(Default, Deserialize)]
+pub(crate) struct Agents {
+    #[serde(default)]
+    ha: Ha,
+}
+
+impl Agents {
+    /// This is a getter for state of the 'ha' feature (enabled/disabled).
+    pub(crate) fn ha_is_enabled(&self) -> bool {
+        self.ha.enabled()
+    }
+}
+
+/// This is used to deserialize the yaml object 'agents.ha'.
+#[derive(Deserialize)]
+pub(crate) struct Ha {
+    #[serde(default = "default_ha_enabled")]
```

Comment:
Is this needed? Seems like we've overriding the helm chart rather than take whatever is there?

Reply:
The helm value may be missing, this will fail upgrade, as it is now. This override, and the others ensure that there is a default value for ha_is_enabled, when one or more the yaml leaf or branches is missing.

Reply:
It could be. No particular reason. All of the defaults only work if it is missing. Would you like it to fail instead (i.e. remove the defaults)?